jerry-core/ecma/builtin-objects/ecma-builtin-function-prototype.cpp

```diff
@@ -226,7 +226,151 @@ ecma_builtin_function_prototype_object_bind (ecma_value_t this_arg, /**< this ar
                                              const ecma_value_t* arguments_list_p, /**< list of arguments */
                                              ecma_length_t arguments_number) /**< number of arguments */
 {
-  ECMA_BUILTIN_CP_UNIMPLEMENTED (this_arg, arguments_list_p, arguments_number);
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 2. */
+  if (!ecma_op_is_callable (this_arg))
+  {
+    ret_value = ecma_make_throw_obj_completion_value (ecma_new_standard_error (ECMA_ERROR_TYPE));
+  }
+  else
+  {
+    /* 4. 11. 18. */
+    ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_FUNCTION_PROTOTYPE);
+    ecma_object_t *function_p = ecma_create_object (prototype_obj_p, true, ECMA_OBJECT_TYPE_BOUND_FUNCTION);
+
+    ecma_deref_object (prototype_obj_p);
+
+    /* 7. */
+    ecma_property_t *target_function_prop_p;
+    target_function_prop_p = ecma_create_internal_property (function_p,
+                                                            ECMA_INTERNAL_PROPERTY_BOUND_FUNCTION_TARGET_FUNCTION);
+
+    target_function_prop_p->u.internal_property.value = ecma_copy_value (this_arg, false);
```

Comment:
Yes, I mean extracting object pointer with `ecma_get_object_from_value` and storing it to `target_function_prop_p->u.internal_property.value`, using `ECMA_SET_NON_NULL_POINTER`.

By "unboxed compressed pointer" I meant plain compressed pointer, without additional type information, stored in `ecma_value_t`.